src/App.js

```diff
@@ -18,69 +20,146 @@ function App() {
-      if (counter === index) {
-        return { id: i.id, letter: newLetter, color: i.color };
-      } else {
-        return { id: i.id, letter: i.letter, color: i.color };
-      }
-    });
-    setArray(newArray);
-  }
+  const addLetters = (newLetter) => {
+    if (counter < maxCounter) {
+      setCounter(counter => counter + 1);
+
+      const newArray = array.map((i, index) => {
+        if (counter === index && i.letter === '') {
+          return { id: i.id, letter: newLetter, color: 'letter-box-gray' };
+        } else {
+          return { id: i.id, letter: i.letter, color: i.color };
+        }
+      });
+      setArray(newArray);
+    }
+  };
 
   // 3. Create function for deleting letters
   const deleteLetters = () => {
-    setCounter(newCounter => newCounter - 1);
+    if (counter > minCounter) {
+      setCounter(newCounter => newCounter - 1);
 
-    const newArray = array.map((i, index) => {
-      if (counter - 1 === index) {
-        return { id: i.id, letter: '', color: 'letter-box' };
-      } else {
-        return { id: i.id, letter: i.letter, color: i.color };
-      }
-    });
-    setArray(newArray);
+      const newArray = array.map((i, index) => {
+        if (counter - 1 === index) {
+          return { id: i.id, letter: '', color: 'letter-box' };
+        } else {
+          return { id: i.id, letter: i.letter, color: i.color };
+        }
+      });
+      setArray(newArray);
+    }
   };
 
-  // 4. Create function for changing button colors
+  // 4. Create function for changing letter colors
   const changeColor = (indexButton) => {
-    const newArray = array.map(i => {
-      if (i.color === 'letter-box' && indexButton === i.id) {
-        return { id: i.id, letter: i.letter, color: 'letter-box-yellow' };
-      } else if (i.color === 'letter-box-yellow' && indexButton === i.id) {
-        return { id: i.id, letter: i.letter, color: 'letter-box-green' };
-      } else if (i.color === 'letter-box-green' && indexButton === i.id) {
-        return { id: i.id, letter: i.letter, color: 'letter-box' };
-      } else {
-        return { id: i.id, letter: i.letter, color: i.color };
-      }
-      ;
-    });
-    setArray(newArray);
+    if (indexButton >= minCounter) {
+      const newArray = array.map(i => {
+        if (i.color === 'letter-box-gray' && indexButton === i.id) {
+          return { id: i.id, letter: i.letter, color: 'letter-box-yellow' };
+        } else if (i.color === 'letter-box-yellow' && indexButton === i.id) {
+          return { id: i.id, letter: i.letter, color: 'letter-box-green' };
+        } else if (i.color === 'letter-box-green' && indexButton === i.id) {
+          return { id: i.id, letter: i.letter, color: 'letter-box-gray' };
+        } else {
+          return { id: i.id, letter: i.letter, color: i.color };
+        };
+      });
+      setArray(newArray);
+    }
   };
 
-  // 5. Create function for 
+  // 5. Create function for submitting word
+  const submit = () => {
```

Comment:
woof - this is a read. 

Once again, drop `let`! 

Add some comments, split this apart. You should really have more interfaces (functional inputs + outputs). 

I'd recommend pulling this function apart into smaller functions and passing inputs to clearly named functions.

```ts

const filterOnLetters = (currentArray, words) => {
const grayLetters = ...;
const yellowLetters = ...;
const greenLetters = ...;
const wordsWithoutGray = wordsWithoutLetters(words, grayLetters);
const wordsWithYellow = wordsWithLetters(wordsWithoutGray, yellowLetters);
const potentialWords = wordsWithLettersInPosition(wordsWithYellow, greenLetters)
 
return potentialWords;
}
```

Also, dealing with unwanted states FIRST is helpful. This is an early return pattern.

```ts
 if (counter !== maxCounter) {
   // nothing happens
   return
   }
```